src/EditorFeatures/CSharpTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.cs

```diff
@@ -1974,6 +1974,43 @@ public async Task TestMetadataOverride()
 }");
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
+        public async Task IgnoreIfTopLevelNullableIsDifferent()
```

Comment:
1. can we have the reverse check where the setter allows nullability, but the getter doesn't?
2. do we have checks that the getter/setter nullability matches the field?  It would need to, right?

Reply:
1. Good call
2. In the getter case, the code would be proper or not already. We wouldn't change that fact. For setter, we only offer if there's a getter being offered from my understanding of the code...

Reply:
That's an interesting point, and I wonder if we want a comparer to encapsulate that logic or just a one off method. There's some notion of permissiveness around nullability, and you could compare to being equal types as long as type1 is at least as permissive as type2 for nullability. The closest thing I can think of is type public/internal/private

Reply:
> In the getter case, the code would be proper or not already.

How so?

Say i have this:

```c#
string foo;
public string? GetFoo() => foo;
```

Is it not wrong if we convert that to `string Foo { get; }` and wrong if we convert it to `string? Foo { get; }`.  Some consume of the existing field or method will break right?

Reply:
That's true, but as implemented we don't change the return type based on the expression. Arguably we should, but right now we don't. 

So, with a close lens on what could happen today, we won't produce code any more invalid than it already was. 

In the case for 

```C#
string foo;
public string? GetFoo() => foo;
```

We should generate

```C#
string foo;
public string? Foo => foo;
```

I'll happily add a test to showcase that behavior.

Reply:
yes please :)  at teh very least, so we don't break tings in the future with the upcomign nullability changes.